dart/lib/src/transport/rate_limit_category.dart

```diff
@@ -0,0 +1,53 @@
+enum RateLimitCategory {
+  all,
+  rate_limit_default, // default
+  error,
+  session,
+  transaction,
+  attachment,
+  security,
+  unknown
+}
+
+extension RateLimitCategoryExtension on RateLimitCategory {
+  static RateLimitCategory fromStringValue(String stringValue) {
+    switch (stringValue) {
+      case '__all__':
+        return RateLimitCategory.all;
+      case 'default':
+        return RateLimitCategory.rate_limit_default;
+      case 'error':
+        return RateLimitCategory.error;
+      case 'session':
+        return RateLimitCategory.session;
+      case 'transaction':
+        return RateLimitCategory.transaction;
+      case 'attachment':
+        return RateLimitCategory.attachment;
+      case 'security':
+        return RateLimitCategory.security;
+    }
+    return RateLimitCategory.unknown;
+  }
+
+  String toStringValue() {
```

Comment:
should we override `toString` instead?

Reply:
`RateLimitCategory` is an enum and you can't override `toString` on an enum :( So this extension is the next best solution.